src/backup/BackupComplete.tsx

```diff
@@ -22,14 +22,17 @@ import fontStyles from 'src/styles/fonts'
 type Props = NativeStackScreenProps<StackParamList, Screens.BackupComplete>
 
 function BackupComplete({ route }: Props) {
-  const navigatedFromSettings = route.params?.navigatedFromSettings ?? false
+  const settingsScreen = route.params?.settingsScreen ?? undefined
   const backupCompleted = useSelector(backupCompletedSelector)
   const { t } = useTranslation()
 
   useEffect(() => {
     const timer = setTimeout(() => {
-      if (navigatedFromSettings) {
-        navigate(Screens.Settings, { promptConfirmRemovalModal: true })
+      if (settingsScreen) {
```

Comment:
might be worth leaving a cleanup comment here. I liked your naming on `isAccountRemoval`, maybe we can do that when we clean the drawer ? Should be also fine to just describe this in the linear issue @MuckT created.